pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -505,34 +505,40 @@ static void validateTaskConfigs(TableConfig tableConfig, Schema schema) {
    *  - comparison column exists
    */
   @VisibleForTesting
-  static void validateUpsertConfig(TableConfig tableConfig, Schema schema) {
-    if (tableConfig.getUpsertMode() == UpsertConfig.Mode.NONE) {
+  static void validateUpsertAndDedupConfig(TableConfig tableConfig, Schema schema) {
+    if (tableConfig.getUpsertMode() == UpsertConfig.Mode.NONE && tableConfig.getDedupConfig() == null) {
```

Comment:
Non-null dedup config doesn't mean it is enabled. We either remove the `dedupEnabled` field and treat non-null dedup as dedup-enabled, or check the flag.

Reply:
Understood. We should also skip the validation when `DedupConfig` is available, but dedup is not enabled

Reply:
Ack